### PR TITLE
CLOUDSTACK-10227: Stabilization fixes for 4.11.0.0

### PR DESCRIPTION
**Pull request by rhtyd**
- Test and regression fixes
- Packaging/systemvmtemplate fixes
- DB upgrade path related fixes
- Upgrade and post-upgrade failure fixes

Signed-off-by: Rohit Yadav <rohit.yadav@shapeblue.com>

### CODE REVIEW

**Comment by rhtyd** [comment]
Outstanding failures:

Vmware:
test_08_migrate_vm | Error | 64.72 | test_vm_life_cycle.py
test_01_create_volume | Failure | 217.41 | test_volumes.py

Xenserver:
test_09_destroy_ssvm | Failure | 113.09 | test_ssvm.py
test_02_edit_template | Failure | 90.22 | test_templates.py

File: packaging/centos63/cloud.spec [hunk 1]
```diff
@@ -130,8 +130,6 @@ Requires: perl
 Requires: libvirt-python
 Requires: qemu-img
 Requires: qemu-kvm
-Requires: epel-release
```
[the remark this answers is not in the document]
Comment (rhtyd) [reply packaging/centos63/cloud.spec]: Thanks Daan, rest of the changes are around fixing smoketests regressions and upgrade related fixes.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@rafaelweingartner @wido @marcaurele and others - review requested. I'll kick smoketests as soon as packaging is back. Also, I'll update checksums of the new 4.11 systemvmtemplate over the weekend in the db upgrade path.

**Comment by rhtyd** [comment]
@blueorangutan test matrix

**Comment by rhtyd** [comment]
@blueorangutan test centos7 xenserver-65sp1

**Comment by wido** [comment]
I find it hard to review this PR as I never have touched VMWare nor the new HA code.

Looking at the code itself I don't see anything weird

**Comment by DaanHoogland** [comment]
rekick

**Comment by DaanHoogland** [comment]
jenkins do your thing

**Comment by DaanHoogland** [comment]
@rhtyd  it always seems to be test_ha_configure_enabledisable_across_clusterzones that fails on travis lately. Any idea what changed with this test? it seems not related to any of the 4.11 features. any platform used by the travis jobs maybe?

**Comment by rhtyd** [comment]
@dahn this PR fixes a NPE case in hostha bg thread, that fixes that failing hostha-simulator test case on Travis. As you can find that with the npe fix, Travis job passes now.

**Comment by rhtyd** [comment]
XenServer tests LGTM. I checked the failures were caused due to test_public_ip_range.py failing to remove a fake public IP range dedicated to systemvms, the SSVM got a public IP that was not reachable and failed iso/template/volume related download cases.

**Comment by rhtyd** [comment]
FYI - I've updated the checksums, built and uploaded 4.11 systemvmtemplates at:
https://download.cloudstack.org/systemvm/4.11/
mirrored also at: http://dl.rohityadav.cloud/testing/

**Comment by rhtyd** [comment]
Merging this based on test results and two lgtms.

**Pull request merged**